src/pkg/mcp/actions/setAWSBYOCProvider.go

```diff
@@ -0,0 +1,102 @@
+package actions
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/DefangLabs/defang/src/pkg/cli/client"
+	"github.com/DefangLabs/defang/src/pkg/mcp/common"
+)
+
+// Check if the provided AWS access key ID is valid
+// https://medium.com/@TalBeerySec/a-short-note-on-aws-key-id-f88cc4317489
+func IsValidAWSKey(key string) bool {
+	// Define accepted AWS access key prefixes
+	acceptedPrefixes := map[string]bool{
+		"ABIA": true,
+		"ACCA": true,
+		"AGPA": true,
+		"AIDA": true,
+		"AKPA": true,
+		"AKIA": true,
+		"ANPA": true,
+		"ANVA": true,
+		"APKA": true,
+		"AROA": true,
+		"ASCA": true,
+		"ASIA": true,
+	}
+
+	if len(key) < 16 {
+		return false
+	}
+
+	prefix := key[:4]
+	_, ok := acceptedPrefixes[prefix]
+	if !ok {
+		return false
+	}
+
+	return true
+}
+
+func SetAWSByocProvider(ctx context.Context, providerId *client.ProviderID, cluster string, awsID string, awsSecret string, region string) error {
```

Comment:
bad param names, the "aws" prefix can be dropped because of func name, so `accessKeyId` and `secretAccessKey` match the env vars.